pkg/provider/azure_controller_common.go

```diff
@@ -337,9 +351,38 @@ func (c *controllerCommon) attachDiskBatchToNode(ctx context.Context, subscripti
 			}
 		}
 
-		err = vmset.WaitForUpdateResult(resultCtx, future, resourceGroup, "attach_disk")
+		backoffManager := wait.Backoff{
+			Duration: updateVMUpdateRetryDuration,
+			Factor:   updateVMUpdateRetryFactor,
+			Steps:    updateVMUpdateRetrySteps,
+		}
+
+	updateRetryLoop:
+		for {
+			if future != nil {
+				err = vmset.WaitForUpdateResult(resultCtx, future, resourceGroup, "attach_disk")
+				updateRequired = c.vmUpdateRequired(future, err)
+			}
+
+			if updateRequired {
+				select {
+				case <-resultCtx.Done():
+					err = context.DeadlineExceeded
+					break updateRetryLoop
+				case <-time.After(backoffManager.Step()):
+				}
+				klog.V(5).Infof("Retry VM Update on node (%s) due to error (%v)", nodeName, err)
```

Comment:
This should be a rare case and it would useful to have this trace to explain unexpected latency and why the previously logged error didn't result in a failed attach. `V(2)` would be appropriate.